gemini/use-cases/education/resume_generator.ipynb

```diff
@@ -0,0 +1,877 @@
+        "    prompt = f\"\"\"\n",
+        "    Extract the following candidate information from the given resume\n",
+        "    1. Name\n",
+        "    2. Address\n",
+        "    3. Mobile Number\n",
+        "    4. Email address\n",
+        "    Text : {resume}\n",
+        "\n",
+        "    INSTRUCTIONS:\n",
+        "    - In case the information is not present, return 'Not Found'\n",
+        "    - The final output should be in valid JSON format\n",
+        "    \"\"\"\n",
+        "    resp = model.generate_content(prompt).text\n",
+        "    return literal_eval(resp.split(\"```json\")[-1].split(\"```\")[0].replace(\"\\n\", \"\"))"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "metadata": {
+        "id": "e87930b0-a15b-4ad7-86fd-a941cebc1637"
+      },
+      "outputs": [],
+      "source": [
+        "def read_pdf(file_path: str) -> str:\n",
+        "    \"\"\"Extracts text from a PDF.\n",
+        "\n",
+        "    Args:\n",
+        "        file_path: path to the pdf file\n",
+        "\n",
+        "    Returns:\n",
+        "        resume content as plain text\n",
+        "    \"\"\"\n",
+        "    with open(file_path, \"rb\") as file:\n",
+        "        reader = pypdf.PdfReader(file)\n",
+        "        text = \"\"\n",
+        "        for page in reader.pages:\n",
+        "            text += page.extract_text()\n",
+        "    return text"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "metadata": {
+        "id": "d2b768d1-afe2-4bf3-88c5-9d27911c0729"
+      },
+      "outputs": [],
+      "source": [
+        "def get_info(resume: str) -> str:\n",
+        "    \"\"\"\n",
+        "    Get the name, address, mobile number and email address from the resume\n",
+        "\n",
+        "    Args:\n",
+        "        resume: The parsed resume as str\n",
+        "\n",
+        "    Returns:\n",
+        "        Details requested in the prompt as string\n",
+        "    \"\"\"\n",
+        "    prompt = f\"\"\"\n",
+        "    Extract the following candidate information from the given resume\n",
+        "    1. Name\n",
+        "    2. Address\n",
+        "    3. Mobile Number\n",
+        "    4. Email address\n",
+        "    Text : {resume}\n",
+        "\n",
+        "    INSTRUCTIONS:\n",
+        "    - In case the information is not present, return 'Not Found'\n",
+        "    - The final output should be in valid JSON format\n",
+        "    \"\"\"\n",
+        "    resp = model.generate_content(prompt).text\n",
+        "    personal_details = literal_eval(\n",
+        "        resp.split(\"```json\")[-1].split(\"```\")[0].replace(\"\\n\", \"\")\n",
+        "    )\n",
+        "    per_details = \"\"\n",
+        "    for k_detail, v_desc in personal_details.items():\n",
+        "        per_details += k_detail + \": \" + v_desc + \"\\n\"\n",
+        "    return per_details"
```

Comment:
Same as before, re: controlled generation

Reply:
Replaced with GenerationConfig class and added controlled generation across the code